app/models/container_group.rb

```diff
@@ -94,4 +94,18 @@ def disconnect_inv
     self.deleted_on = Time.now.utc
     save
   end
+
+  def self.post_refresh_ems(ems_id, _ems_refresh_start_time)
+    self.reconnect_events(ems_id)
+  end
+
+  def self.reconnect_events(ems_id)
+    events = EmsEvent.where(:ems_id => ems_id, :container_group_id => nil).where.not(:container_group_name => nil)
+    events.each do |event|
```

Comment:
**Performance note:** (but lets figure out other aspects before optimizing)

Above query may return tons of disconnected events, and it's possible that an *ever-increasing* number of them will remain disconnected (since we're missing short-lived pods).
=> Looping over all disconnected events risks making refresh increasingly slow, repeating sisyphean attempts to reconnect unconnectable events...
=> Would be much better as a join only returning *matching* event-pod pairs, or maybe directly reconnecting them in-db with `update_all`.
=> Perhaps needs a time horizon setting to stop look at events older than some time?

Reply:
> => Would be much better as a join only returning matching event-pod pairs, or maybe directly reconnecting them in-db with update_all.

Right.

> => Perhaps needs a time horizon setting to stop look at events older than some time?

how about EmsEvent.where(...).where(ems.last_refresh_date < event.timestamp < Time.now)

Reply:
> Above query may return tons of disconnected events, and it's possible that an ever-increasing number of them will remain disconnected (since we can never see some short-lived pods).

Good point (should improve with the pod watch). I seem to remember there was an infrastructure related to getting all the updated objects after a refresh, so we can know who the candidates for re connection are? @Ladas ? (maybe I'm just making this up)